contracts/RMRK/extension/tokenProperties/RMRKTokenPropertiesRepository.sol

```diff
@@ -409,6 +429,224 @@ contract RMRKTokenPropertiesRepository is IRMRKTokenPropertiesRepository {
+            ] = properties[i].value;
+            emit UintPropertyUpdated(
+                collection,
+                tokenId,
+                properties[i].key,
+                properties[i].value
+            );
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /**
+     * @inheritdoc IRMRKTokenPropertiesRepository
+     */
+    function setBoolProperties(
+        address collection,
+        uint256 tokenId,
+        BoolProperty[] memory properties
+    ) external onlyAuthorizedCaller(collection, "", tokenId) {
+        for (uint256 i = 0; i < properties.length; ) {
+            _boolValues[collection][tokenId][
+                _getIdForKey(properties[i].key)
+            ] = properties[i].value;
+            emit BoolPropertyUpdated(
+                collection,
+                tokenId,
+                properties[i].key,
+                properties[i].value
+            );
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /**
+     * @inheritdoc IRMRKTokenPropertiesRepository
+     */
+    function setAddressProperties(
+        address collection,
+        uint256 tokenId,
+        AddressProperty[] memory properties
+    ) external onlyAuthorizedCaller(collection, "", tokenId) {
+        for (uint256 i = 0; i < properties.length; ) {
+            _addressValues[collection][tokenId][
+                _getIdForKey(properties[i].key)
+            ] = properties[i].value;
+            emit AddressPropertyUpdated(
+                collection,
+                tokenId,
+                properties[i].key,
+                properties[i].value
+            );
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /**
+     * @inheritdoc IRMRKTokenPropertiesRepository
+     */
+    function setBytesProperties(
+        address collection,
+        uint256 tokenId,
+        BytesProperty[] memory properties
+    ) external onlyAuthorizedCaller(collection, "", tokenId) {
+        for (uint256 i = 0; i < properties.length; ) {
+            _bytesValues[collection][tokenId][
+                _getIdForKey(properties[i].key)
+            ] = properties[i].value;
+            emit BytesPropertyUpdated(
+                collection,
+                tokenId,
+                properties[i].key,
+                properties[i].value
+            );
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /**
+     * @inheritdoc IRMRKTokenPropertiesRepository
+     */
+    function setProperties(
+        address collection,
+        uint256 tokenId,
+        StringProperty[] memory stringProperties,
+        UintProperty[] memory uintProperties,
+        BoolProperty[] memory boolProperties,
+        AddressProperty[] memory addressProperties,
+        BytesProperty[] memory bytesProperties
+    ) external onlyAuthorizedCaller(collection, "", tokenId) {
+        if (stringProperties.length > 0) {
```

Comment:
This ifs outside the loop are unnecessary. If length 0 the loop will do nothing.